File: pset1/cash/cash.c
//libraries
#include<cs50.h>
#include<stdio.h>
#include<math.h>

int main(void)
{
    // prompt user for positive number
    float i;
    do
    {
        i = get_float("Change owed: ");
    }
    while (i < 0);

    // calculation of optimal solution
    int j = round(100 * i);
    int a = j / 25;
    int b = (j % 25) / 10;
    int c = ((j % 25) % 10) / 5;
    int d = (((j % 25) % 10) % 5) / 1;
    int e = a + b + c + d;

    //print the solution
    printf("%i\n", e);
}

File: pset6/cash/cash.py
from cs50 import get_float

input = -1
while input < 0:
    input = get_float("Change owed: ")
input *= 100
a = input // 25
input = input - a * 25

b = input // 10
input = input - b * 10

c = input // 5
input = input - c * 5

d = input // 1

print(int(a + b + c + d))